modules/rostests/apitests/shell32/ShellExecuteEx.cpp

```diff
@@ -111,15 +111,37 @@ getCommandLineFromProcess(HANDLE hProcess)
     PEB peb;
     PROCESS_BASIC_INFORMATION info;
     RTL_USER_PROCESS_PARAMETERS Params;
+    LONG status;
+    BOOL ret;
+
+    status = NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
+    if (status != 0)
+        trace("status: 0x%08lX\n", status);
+
+    ret = ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
-    NtQueryInformationProcess(hProcess, ProcessBasicInformation, &info, sizeof(info), NULL);
-    ReadProcessMemory(hProcess, info.PebBaseAddress, &peb, sizeof(peb), NULL);
     ReadProcessMemory(hProcess, peb.ProcessParameters, &Params, sizeof(Params), NULL);
+    if (!ret)
+        trace("ReadProcessMemory failed (%ld)\n", GetLastError());
 
     LPWSTR cmdline = Params.CommandLine.Buffer;
+    if (!cmdline)
+        trace("!cmdline\n");
+
     SIZE_T cchCmdLine = Params.CommandLine.Length;
```

Comment:
@learn-more This length is the length of characters. Otherwise, it fails like this:

![test-failed-with-half-length](https://github.com/reactos/reactos/assets/2107452/2167511c-9244-49fb-a73a-d758733d889b)

Reply:
The PEB's user parameters `CommandLine.Length` _**cannot**_ be a length in characters, since it's a `UNICODE_STRING` . The `Length` parameter is in bytes. You can verify this because literally _**all**_ the code in ReactOS that looks at this CommandLine string expects this, and it works.
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/lib/nt/entry_point.c;hb=2f6b17593f310115e48d5baa7dd16c297024ea7a#l102
https://git.reactos.org/?p=reactos.git;a=blob;f=modules/rostests/apitests/msvcrt/CmdLineUtil/CmdLineUtil.c;hb=2f6b17593f310115e48d5baa7dd16c297024ea7a#l1
https://git.reactos.org/?p=reactos.git;a=blob;f=dll/ntdll/ldr/ldrinit.c;hb=2f6b17593f310115e48d5baa7dd16c297024ea7a#l1850
etc. etc. .....

Reply:
Then, why
`SIZE_T cchCmdLine = Params.CommandLine.Length / sizeof(WCHAR);` won't work in Windows?

Reply:
It all depends how you're using cchCmdLine afterwards (in the ReadProcessMemory call mainly)

Reply:
Fixed as `cbCmdLine`.